client/src/GameState.ts

```diff
@@ -22,6 +34,52 @@ export class GameState {
     otherTetrominoes: Array<Tetromino>;
     playerId!: 0 | 1 | 2 | 3;
 
+    lineCheckSequence!: Array<LineCheckTask>;
+    // template used to generate lineCheckSequence once playerId is received
+    static LineCheckTemplate: Array<LineCheckTask> = [
+        {
+            axis: LineCheckAxis.Row,
+            direction: 1,
+            range: [],
+        },
+        {
+            axis: LineCheckAxis.Column,
+            direction: -1,
+            range: [],
+        },
+        {
+            axis: LineCheckAxis.Row,
+            direction: -1,
+            range: [],
+        },
+        {
+            axis: LineCheckAxis.Column,
+            direction: 1,
+            range: [],
+        },
+    ];
+
+    private generateLineCheckSequence(
+        playerId: number = 0
+    ): Array<LineCheckTask> {
```

Comment:
[used also for initializing the sequence](https://github.com/ECE493G14Capstone/Capstone/pull/74/files#diff-af947977ce4c662b2ea46dfd0203d6c76ed28e5c310aad6c0d667d7dbbeab183R139), I was thinking that we can pass in player ID in case it does not exist for spectators. But I think your suggestion can also work by defaulting it to 0 if not exist, so will do that.